scripts/spack/packages/axom/package.py

```diff
@@ -93,11 +93,12 @@ class Axom(CachedCMakePackage, CudaPackage, ROCmPackage):
     # Dependencies
     # -----------------------------------------------------------------------
     # Basics
-    depends_on("cmake@3.14:", type="build")
+    depends_on("cmake@3.14:", type="build", when="@:0.6.1")
+    depends_on("cmake@3.18:", type="build", when="@0.7.0:")
     depends_on("cmake@3.21:", type="build", when="+rocm")
```

Comment:
To keep track of: RAJA, Umpire, and camp require CMake 3.23 for ROCm builds for v2022.10.x releases